File: stvs/src/test/java/edu/kit/iti/formal/stvs/logic/io/xml/XmlConfigExporterTest.java
package edu.kit.iti.formal.stvs.logic.io.xml;

import edu.kit.iti.formal.stvs.model.config.GlobalConfig;
import org.apache.commons.io.IOUtils;
import org.junit.Before;
import org.junit.Test;

import java.io.ByteArrayOutputStream;

import static org.junit.Assert.assertEquals;

/**
 * @author Benjamin Alt
 */
public class XmlConfigExporterTest {

  private XmlConfigExporter exporter;

  @Before
  public void setUp() {
    exporter = new XmlConfigExporter();
  }

  @Test
  public void testExportConstraintDefault() throws Exception {
    GlobalConfig globalConfig = new GlobalConfig();
    globalConfig.setZ3Path("[Path to Z3 Executable]");
    globalConfig.setNuxmvFilename("[Path to NuXmv Executable]");
    ByteArrayOutputStream result = exporter.export(globalConfig);
    String resultString = new String(result.toByteArray(), "utf-8");
    String expectedString = IOUtils.toString(
        this.getClass().getResourceAsStream("config_valid_default.xml"), "UTF-8");
    assertEquals(TestUtils.removeWhitespace(expectedString), TestUtils.removeWhitespace
        (resultString));
  }

}

File: stvs/src/test/kotlin/edu/kit/iti/formal/stvs/logic/io/xml/XmlConfigExporterTest.kt
package edu.kit.iti.formal.stvs.logic.io.xml

import com.google.common.truth.Truth
import edu.kit.iti.formal.stvs.model.config.GlobalConfig
import org.junit.jupiter.api.Test

/**
 * @author Benjamin Alt
 */
class XmlConfigExporterTest {
    private val exporter = XmlConfigExporter()

    @Test
    fun testExportConstraintDefault() {
        val globalConfig = GlobalConfig()
        globalConfig.z3Path = "[Path to Z3 Executable]"
        globalConfig.nuxmvFilename = "[Path to NuXmv Executable]"
        val act = TestUtils.stringOutputStream { exporter.export(globalConfig, it) }
        val exp = javaClass.getResourceAsStream("config_valid_default.xml")!!.bufferedReader().readText()
        Truth.assertThat(
            TestUtils.removeWhitespace(act)
        ).isEqualTo(TestUtils.removeWhitespace(exp))
    }
}